src/capi/azext_capi/custom.py

```diff
@@ -592,33 +593,34 @@ def create_workload_cluster(  # pylint: disable=too-many-arguments,too-many-loca
     logger.warning('✓ Workload access configuration written to "%s"', workload_cfg)
 
     # Install CNI
-    calico_manifest = "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/master/templates/addons/calico.yaml"  # pylint: disable=line-too-long
+    HelmInfo = namedtuple("HelmInfo", ["repo_name", "repo_url", "chart_name", "chart", "values_file", "args"])
+    helminfo = HelmInfo(
+        repo_name="projectcalico",
+        repo_url="https://projectcalico.docs.tigera.io/charts",
+        chart_name="calico",
+        chart="projectcalico/tigera-operator",
+        # TODO: choose a different values files for IPv6 and dual-stack templates
+        values_file="https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/" + \
+                    "main/templates/addons/calico/values.yaml",
+        args=[
+            "--namespace", "tigera-operator", "--create-namespace",
+            '--set-string', 'installation.calicoNetwork.ipPools[0].cidr="$IPV4_CIDR_BLOCK"',
```

Comment:
where is IPV4_CIDR_BLOCK coming from?

Reply:
From the docs? 😄 I see we're not using that in the CAPZ Tiltfile so I'll drop it.

Reply:
in the doc there is a line that sets it first, here it's not being set anywhere. Tilt doesn't set it because it assumes default CIDR from our flavor templates.

Reply:
I removed that argument since we make basically the same assumptions as Tilt, and it's working fine now for Linux clusters. Now I'll implement the Windows configmap workaround and see what's left to do.

Reply:
That won't be resilient when users pass in custom templates with different cluster pod CIDRs, consider passing in the cluster CIDR value from Cluster.Spec so it _always_ works

Reply:
We are now finding and passing through the cluster CIDR value(s) here. 👍🏻